### ISSUE TITLE
dialect: sparksql doesn't support `*` in a multi-argument function call

### ISSUE THREAD

**Issue by spex66** [sqlfluff/sqlfluff#4873]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

This file 

```sql
SELECT
    map_concat(
        map_from_arrays(
            transform(
                -- causing a
                -- Line 79 Position 56: Found unparsable section: ', *'
                map_values(
                    to_metadata_except(
                        array('key', 'lastUpdatedTime'), *
                    )
                ),
                x -> x
            )
        )
    ) AS metadata
FROM something
```

reports an `unparsable section:` when linting

```bash
✗ sqlfluff lint -v --dialect=sparksql sqlfluff-test2.sql
==== sqlfluff ====
sqlfluff:                2.1.0 python:                 3.11.3
implementation:        cpython verbosity:                   1
dialect:              sparksql templater:               jinja
rules:                                all
==== readout ====

== [sqlfluff-test2.sql] FAIL                                                                                                                                                                                            
L:   9 | P:  56 |  PRS | Line 9, Position 56: Found unparsable section: ', *'
WARNING: Parsing errors found and dialect is set to 'sparksql'. Have you configured your dialect correctly?
==== summary ====
violations:        1 status:         FAIL
All Finished 📜 🎉!
```

### Expected Behaviour

Even if `to_metadata_except(..)` is a custom spark-function, `*` should be supported in a multi-argument function call, and not only like in `count(*)`.

### Observed Behaviour

The linter reports the "**Found unparsable section: ', *'**"

Discussed with @tunetheweb on [slack here](https://sqlfluff.slack.com/archives/C01GPHR8J0G/p1684324829174189?thread_ts=1684316652.220449&cid=C01GPHR8J0G) where Barry suggest the fix to make in `dialect_sparksql.py` to make it work.

I'll try to open a PR with the change above today.

This issue is related to #4871 which is about `sqlfluff fix --show-lint-violation` not working as expected.

### How to reproduce

See steps above.

### Dialect

sparksql

### Version

✗ sqlfluff version
2.1.0

➟  python --version
Python 3.11.3

### Configuration

no `.sqlfluff` or `pyproject.toml` section used

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
